src/gazebo_opticalFlow_plugin.cpp

```diff
@@ -146,62 +151,43 @@ void OpticalFlowPlugin::OnNewFrame(const unsigned char * _image,
 #endif
   const float focal_length = (_width/2)/tan(hfov/2);
 
-#if GAZEBO_MAJOR_VERSION >= 7
-  float rate = this->camera->RenderRate();
-#else
-  float rate = this->camera->GetRenderRate();
-#endif
-  //printf("render rate raw: %10.2f\n", rate);
-  if (!isfinite(rate))
-       rate =  30.0;
-  float dt = 1.0 / rate;
-
-
+  timer_.stop();
+  float rate = 100; // assume 100 hz
+  float dt = 1.0/rate;
   Mat frame = Mat(_height, _width, CV_8UC3);
+  Mat frameHSV = Mat(_height, _width, CV_8UC3);
   Mat frameBGR = Mat(_height, _width, CV_8UC3);
   frame.data = (uchar*)_image; //frame is not the right color now -> convert
-  cvtColor(frame, frameBGR, CV_RGB2BGR); 
+  // dynamically scale image using HSV full conversion
+  cvtColor(frame, frameHSV, CV_RGB2HSV_FULL); 
+  cvtColor(frameHSV, frameBGR, CV_HSV2BGR); 
   cvtColor(frameBGR, frame_gray, CV_BGR2GRAY);
-  
-  featuresPrevious = featuresCurrent;
-
-  goodFeaturesToTrack(frame_gray, featuresCurrent, maxfeatures, qualityLevel, minDistance, Mat(), blockSize, useHarrisDetector, k); //calculate the features
-
-  if (!old_gray.empty() && featuresPrevious.size() > 0){
-    calcOpticalFlowPyrLK(old_gray, frame_gray, featuresPrevious, featuresNextPos, featuresFound, err, cv::Size(8,8), 0);
+  float x_gyro_rate = 0;
+  float y_gyro_rate = 0;
+  float z_gyro_rate = 0;
+  float x_flow = 0;
+  float y_flow = 0;
+
+  // check for correct image size
+  if (frame_gray.rows != 64 || frame_gray.cols != 64) {
```

Comment:
This check should go into the Load() method, as the size is not changed after startup.